packages/node_modules/@webex/plugin-memberships/src/memberships.js

```diff
@@ -415,7 +500,47 @@ const Memberships = SparkPlugin.extend({
       body: membership
     })
       .then((res) => res.body);
+  },
+
+  /**
+   * Updates the lastSeenId attribute of a membership
+   * @instance
+   * @memberof Mememberships
+   * @param {string} message
+   * @returns {Promise<MembershipObject>}
+   */
+  updateLastSeen(message) {
+    const activity = {
+      id: deconstructHydraId(message.id).id
+    };
+    const conversation = {
+      id: deconstructHydraId(message.roomId).id
+    };
+
+    return this.spark.internal.conversation.acknowledge(conversation, activity)
+      .then((activity) => {
+        console.log(activity);
+
+        return ({
+          lastSeenId: constructHydraId(hydraTypes.MESSAGE, activity.object.id),
+          id: constructHydraId(hydraTypes.MEMBERSHIP,
+            `${activity.actor.entryUUID}:${activity.target.id}`),
+          personId: constructHydraId(hydraTypes.PEOPLE,
+            activity.actor.entryUUID),
+          personEmail: activity.actor.emailAddress || activity.actor.entryEmail,
+          personDisplayName: activity.actor.displayName,
+          personOrgId: constructHydraId(hydraTypes.ORGANIZATION,
+            activity.actor.orgId),
+          roomId: constructHydraId(hydraTypes.ROOM, activity.target.id),
+          roomType:
+            activity.target.tags.includes(SDK_EVENT.INTERNAL.ACTIVITY_TAG.ONE_ON_ONE) ?
+              SDK_EVENT.EXTERNAL.SPACE_TYPE.DIRECT :
+              SDK_EVENT.EXTERNAL.SPACE_TYPE.GROUP,
+          isRoomHidden: false // any activity unhides a space.
```

Comment:
It might be a tad cleaner to have a function that you pass these objects too and based on keys, it properly encodes the things it needs to.

Reply:
Do mean 
```
return ({
  key1 = makeKey1(activity),
  key2 = makeKey2(activity)
});
```
or
```
 return(convertActivityToMembershipSeen(activity));

Reply:
I gave a cleanup technique a shot if you want to check it out again